File: tests/ts/pack_unpack_simple.test.ts
import { test } from 'bun:test';
import { ObjS } from '../dist/ts/obj_ObjS';

test('pack and unpack data', () => {
    const obj = new ObjS({
        f64: -0.01,
    });
    const s1 = JSON.stringify(obj, (_key, value) => {
        return typeof value === 'bigint' ? value.toString() : value;
    });
    console.debug('Original object JSON:', s1);
    const packed = obj.pack();
    console.debug('Packed data length:', packed);
    const [unpacked, err] = ObjS.unpack(packed);
    if (err) {
        throw err;
    }
    const s2 = JSON.stringify(unpacked, (_key, value) => {
        return typeof value === 'bigint' ? value.toString() : value;
    });
    console.debug('Unpacked object JSON:', s2);
});

File: tests/pack_unpack_simple_test.go
package tests

import (
	"encoding/json"
	"fmt"
	"testing"

	m "github.com/bbanez/minibin/tests/dist/go"
)

func TestPackSimple(t *testing.T) {
	obj := m.NewObjS(
		-10.1234,
	)
	s1, err := json.Marshal(obj)
	if err != nil {
		t.Fatal(err)
	}
	packed := obj.Pack()
	fmt.Println("Packed", packed)
	unpacked, err := m.UnpackObjS(packed, nil)
	if err != nil {
		t.Fatal(err)
	}
	s2, err := json.Marshal(unpacked)
	if err != nil {
		t.Fatal(err)
	}
	fmt.Println("S1", string(s1))
	fmt.Println("S2", string(s2))
}
